### PR TITLE
Start work on alternative client

### PR DESCRIPTION
**Pull request by lubomir**
This client will have nicer user interface. The client allows loading
plugins from predefined directories so that implementing internal
extensions is possible.

#### TODO
 * [x] Split help into user help (`--help`) and admin help (`--help-all`).
 * [x] Figure out a way to split the code into multiple modules.
 * [ ] Add unit tests.

### CODE REVIEW

File: pdc_client/bin/pdc [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+import os.path
+import warnings
+import itertools
+import logging
+import imp
+import argcomplete
+
+
+PLUGIN_DIRS = [
+    os.path.join(os.path.dirname(__file__), '..', 'plugins')
```
Comment (erichuanggit) [inline pdc_client/bin/pdc]: what's `plugins` for?

File: requirements/production.txt [hunk 2]
```diff
@@ -12,3 +12,5 @@ django-filter==0.9.2
 django-mptt>=0.7.1
 beanbag>=1.9.2
 django-cors-headers
+argparse
```
Comment (erichuanggit) [inline requirements/production.txt]: I think both of them need to be added in pdc_client.spec as well.

File: pdc_client/bin/pdc (hunk 1 above)
Comment (lubomir) [inline pdc_client/bin/pdc]: Supposing the client should be open-source, we need some way to extend it to work with internal parts. This was the easiest way to achieve that. The client finds all python modules in a predefined place, loads the modules and later calls functions in them (`run_hook`). The internal things would just implement the module, drop it into proper place and the client would just work.

File: requirements/production.txt (hunk 2 above)
Comment (lubomir) [inline requirements/production.txt]: Correct, but argparse is actually in standard lib, so it should be removed.

**Comment by erichuanggit** [comment]
I like this code structure, loose couple and easy to maintain.

**Comment by lubomir** [comment]
I have rebased the patches on master and restructured the commits so that they are meaningful. It is on pull request #125. So far, I don't have a clear idea on how to reasonably test the code.

**Pull request closed without merging**